### ISSUE TITLE
Different qualities not listed for some gems

### ISSUE THREAD

**Issue by FireAllianceNX** [Centias/BankItems#17]
Working on some late-expansion jewelcrafting and noticed that some gems are not listing all of its qualities and count (or listing incorrectly). I am using the mouseover off the crafting window for most of these tests (over actual item seem to reproduce the same issue)

For example (not an exhausting list):
For _Neltharite_, I have 0 one-star , 27 two-star, 36 three-star, and some random on other alts
Tooltip over the one-star will show no Neltharite (but showing a 2 star on another toon)
Tooltip over the two-star will _only_ show the 27 2-stars (and the 2 star on another toon)
Tooltip over the three-star will show 1-star with 27, 3-stars with 36 (and the other toon is now showing 1 star of this gem)

For _Alexstraszite_, 2 one-star, 24 two-star, 54 three-star.
Tooltip on one-star shows: 2 one-star, 24 two-star, no three-star
Tooltip on two-star shows same as above
Tooltip on three-star shows: 24 one-star, 54 three-star

No problem on _Nozdorite_ or _Ysemerald_, shows all 3 on each quality gem.
On mouse-over in crafting window where you can pick which quality gem to use (before you click), _Malygite_ doesn't display any info at all where as the other 2 above does correctly. (Now, they're not all in my bags even though it says it is, but that doesn't matter as much to me since we can craft with things in the bank, happy with just a total, which seems to be the right total for these two gems)
The 2 that comes back wrong will also display the wrong tooltip here saying only 1 two-star gem is in my bags.

Attached some screenshot in case my wording is hard to understand.

![WoWScrnShot_082123_192902](https://github.com/Centias/BankItems/assets/5430444/67233e99-8530-4bc4-9337-e98169782951)
![WoWScrnShot_082123_192959](https://github.com/Centias/BankItems/assets/5430444/47e3a7c2-ed90-4993-b711-6e001a7e05ca)
![WoWScrnShot_082123_193007](https://github.com/Centias/BankItems/assets/5430444/3ce9f730-49ad-4b26-95fc-54398461426d)

**Comment by Centias** [comment]
I'll have to get back to this soon, but it looks like the same problem as some of the ore that came up before.

Alextraszite : 
rank 1 - 192852
rank 2 - 192853
rank 3 - 192855

Neltharite :
rank 1 - 192862
rank 2 - 192863
rank 3 - 192865

**Comment by Centias** [comment]
[BankItems.zip](https://github.com/Centias/BankItems/files/12413980/BankItems.zip)

Slapped together a quick fix for Alexstraszite and Neltharite, I'll see if I can do anything about the rest and put out an actual update later.

**Comment by Centias** [comment]
Pushed an update with the fix for those gems, though I think I sort of see what is going on with the crafting window tooltips and will have to dig into that again.

**Comment by Centias** [comment]
I think I found a fix for the other problem: The crafting window might be showing quality 2 if you don't have quality 1, but _technically_ the button actually has the ID for quality 1 always. If you don't have any of quality 1 on the current character, it would skip over building the tooltip info even if you had quality 2 or 3 available. 

I changed the check to see if the character has the item you're looking at, OR quality 1 of that item, OR quality 2, OR quality 3, so it builds the line for the current character if any is true.
